src/query_ctrl.ts

```diff
@@ -80,6 +81,16 @@ export class CogniteQueryCtrl extends QueryCtrl {
     });
   }
 
+  onChangeCustom() { // todo: why it doesn't work without this hack?
+    try {
+      parse(this.target.expr);
```

Comment:
this will not work cause it doesn't trigger `$digest` circle to render the component. If we want to handle errors for STS, we should use `$scope` params, i.e. change `$scope` params

Reply:
oh, actually it works. So what's the hack here?

Reply:
the things is that we have error handling deeper in the code, like in `handleFailedTargets`, but when I try assign errors to target by doing `target.error='message'` it doesn't work. And that's actually preferred way for me, since then we we can show all other errors, like http. but it doesn't display the error for some reason.
I though maybe you could find the reason (it's very angular specific for me)

Reply:
Here is a thing, that we can't control rendering from the `query` method. What we can do here is to use `appEvents` grafana event emitter. I'll come up with a solution tmrw.

Reply:
we decided to do this as a separate task